src/Shared/TaskHostConfiguration.cs

```diff
@@ -417,7 +422,26 @@ public void Translate(ITranslator translator)
             translator.TranslateCulture(ref _culture);
             translator.TranslateCulture(ref _uiCulture);
 #if FEATURE_APPDOMAIN
-            translator.TranslateDotNet(ref _appDomainSetup);
+            if (ChangeWaves.AreFeaturesEnabled(ChangeWaves.Wave17_10) || !Traits.Instance.EscapeHatches.IsBinaryFormatterSerializationAllowed)
```

Comment:
Is it really intended to use `||` here? I.e. opting into the previous behavior requires disabling the change-wave **and** setting the escape hatch?

Reply:
I was following https://github.com/dotnet/msbuild/pull/9318/files#diff-39cbfa4c8cd24d6a641a794b405985440b851ee5ff3ba29c98af486227e79b39R589 as I was told to. I'm not sure if it's intention though.

Reply:
Tagging @JanKrivanek to confirm.

Reply:
All good - this is inline with other related changeways. It's to prevent users from shooting into their foot by opting into BinFmt behavior that'd lead to runtime error.

Reply:
Thank you for confirming!

Reply:
Please document the changewave (along with the discussed caveat) in the ChangeWaves.md file. Similarly to:

https://github.com/dotnet/msbuild/pull/9262/files#diff-854a463890211b66f6c60646fe5d85736e82b0daaa2152561a055b333db480a8R30